host_core/lib/host_core/providers/provider_supervisor.ex

```diff
@@ -133,6 +133,7 @@ defmodule HostCore.Providers.ProviderSupervisor do
     |> Enum.map(fn {_d, pid, _type, _modules} ->
       provider_for_pid(pid)
     end)
+    |> Enum.filter(fn provider -> provider != nil end)
```

Comment:
As an FYI for the future, there's a function in the kernel called `is_nil` and you can use `reject` to reject the nils, making this a bit easier to read: 
```
Enum.reject(list, &is_nil/1)
```

Reply:
I actually like this a lot better, going to go ahead and implement it here